flake.nix

```diff
@@ -98,6 +98,23 @@
             with pkgs.lib.attrsets;
             mapAttrs' (name: value: nameValuePair (s + name) value) attrs;
 
+          addWerror = x: x.override { ghcOptions = [ "-Werror" ]; };
+
+          componentsToWerrors = n: x:
+            builtins.listToAttrs
+              ([
+                {
+                  name = "${n}-werror";
+                  value = addWerror x.components.library;
+                }
+              ]) // lib.attrsets.mergeAttrsList (map
+              (y:
+                lib.mapAttrs'
+                  (k: v: {
+                    name = "${n}-${y}-${k}-werror";
+                    value = addWerror v;
+                  })
+                  x.components."${y}") [ "benchmarks" "exes" "sublibs" "tests" ]);
```

Comment:
Could: (I think) use `collectComponents` function from haskell.nix for this https://input-output-hk.github.io/haskell.nix/reference/library.html#collectcomponents-collectcomponents